src/Umbraco.Core/Cache/AppCacheExtensions.cs

```diff
@@ -43,24 +43,41 @@ public static void InsertCacheItem<T>(
     public static T? GetCacheItem<T>(this IAppCache provider, string cacheKey)
     {
         var result = provider.Get(cacheKey);
-        if (IsRetrievedItemNull(result))
+        if (result == null)
         {
             return default;
         }
 
+        // If we've retrieved the specific string that represents null in the cache, return it only if we are requesting it (via a typed request for a string).
```

Comment:
I'm not sure I understand this. If I read it correctly:

1. Given `T` is `string`, and
2. The `string` item identified by `cacheKey` does not exist, and
3. `null` values are being cached

...we return the value of `Cms.Core.Constants.Cache.NullRepresentationInCache` (`"*NULL*"`) to whomever requested the `string` for `cacheKey`. 

I'd expect a `null` value returned in all and every cases where no item could be identified `cacheKey` - no matter if we're caching `null` values or not.

Perhaps I'm missing something here?